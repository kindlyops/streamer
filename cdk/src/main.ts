import * as glue from '@aws-cdk/aws-glue';
import * as iam from '@aws-cdk/aws-iam';
import * as kinesis from '@aws-cdk/aws-kinesis';
import * as firehose from '@aws-cdk/aws-kinesisfirehose';
import * as logs from '@aws-cdk/aws-logs';
import * as s3 from '@aws-cdk/aws-s3';
import * as cdk from '@aws-cdk/core';

function String(name: string): glue.Column {
    return {
        name: name,
        type: glue.Schema.STRING,
    };
}

function Int(name: string): glue.Column {
    return {
        name: name,
        type: glue.Schema.INTEGER,
    };
}

export class KinesisTestStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props: cdk.StackProps = {}) {
    super(scope, id, props);

    const bucket = new s3.Bucket(this, 'FirehoseDeliveryBucket',  {
      autoDeleteObjects: true,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      enforceSSL: true,
      encryption: s3.BucketEncryption.S3_MANAGED,
    });

    const glueDb = new glue.Database(this, 'KinesisTestGlueDatabase', {
      databaseName: 'kinesis_test',
    });

    const tableName = 'test_glue_table';

    new glue.Table(this, 'ApplogsGlueTable', {
            bucket: bucket,
            s3Prefix: 'kinesis_delivered/',
            database: glueDb,
            tableName: tableName,
            dataFormat: glue.DataFormat.PARQUET,
            columns: [
                String('foo'),
                Int('n'),
                Int('baz'), // this one we will generate data conversion errors with
            ],
            partitionKeys: [String('ingesteddate')],
        });

    const deliveryRole = new iam.Role(this, 'DeliveryRole', {
            assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
            externalIds: [cdk.Aws.ACCOUNT_ID],
        });

    const source = new kinesis.Stream(this, 'IngestStream', {
            streamName: 'kinesis-test-ingest-stream',
            shardCount: 1,
            retentionPeriod: cdk.Duration.days(1),
            encryption: kinesis.StreamEncryption.MANAGED,
        });

    source.grantRead(deliveryRole);

    deliveryRole.addToPolicy(
            new iam.PolicyStatement({
                resources: [source.streamArn],
                actions: [
                    'kinesis:DescribeStream',
                    'kinesis:GetShardIterator',
                    'kinesis:GetRecords',
                    'kinesis:ListShards',
                    'firehose:DeleteDeliveryStream',
                    'firehose:PutRecord',
                    'firehose:PutRecordBatch',
                    'firehose:UpdateDestination',
                ],
            })
        );

    deliveryRole.addToPolicy(
        new iam.PolicyStatement({
            resources: ['*'],
            actions: ['glue:GetTable', 'glue:GetTableVersion', 'glue:GetTableVersions'],
        })
    );

    deliveryRole.addToPolicy(
        new iam.PolicyStatement({
            resources: [`${bucket.bucketArn}/*`, bucket.bucketArn],
            actions: [
                's3:AbortMultipartUpload',
                's3:GetBucketLocation',
                's3:GetObject',
                's3:ListBucket',
                's3:ListBucketMultipartUploads',
                's3:PutObject',
            ],
        })
    );

    const logGroupName = `/aws/kinesisfirehose/kdf-${tableName}`;

    deliveryRole.addToPolicy(
        new iam.PolicyStatement({
            resources: [
                `arn:aws:logs:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:log-group:${logGroupName}:log-stream:*`,
            ],
            actions: ['logs:PutLogEvents'],
        })
    );

    const logGroup = new logs.LogGroup(this, 'FirehoseLogGroup', {
        retention: logs.RetentionDays.ONE_WEEK,
        logGroupName: logGroupName,
        removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    const logStream = new logs.LogStream(this, 'FirehoseLogStream', {
        logGroup: logGroup,
        removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    const backupsLogStream = new logs.LogStream(this, 'BackupsLogStream', {
        logGroup: logGroup,
        removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    const delivery = new firehose.CfnDeliveryStream(this, 'FirehoseDelivery', {
            deliveryStreamType: 'KinesisStreamAsSource',
            kinesisStreamSourceConfiguration: {
                roleArn: deliveryRole.roleArn,
                kinesisStreamArn: source.streamArn,
            },
            extendedS3DestinationConfiguration: {
                bucketArn: bucket.bucketArn,
                roleArn: deliveryRole.roleArn,
                s3BackupMode: 'Enabled',
                s3BackupConfiguration: {
                    bucketArn: bucket.bucketArn,
                    roleArn: deliveryRole.roleArn,
                    prefix: 'raw_original_records/IngestedDate=!{timestamp:yyyy-MM-dd}/',
                    errorOutputPrefix:
                        'raw_backups_errors/!{firehose:error-output-type}/!{timestamp:yyyy-MM-dd}/',
                    cloudWatchLoggingOptions: {
                        enabled: true,
                        logGroupName:  logGroup.logGroupName,
                        logStreamName: backupsLogStream.logStreamName,
                    },
                },
                bufferingHints: {
                    intervalInSeconds: 60,
                },
                cloudWatchLoggingOptions: {
                    enabled: true,
                    logGroupName: logGroup.logGroupName,
                    logStreamName: logStream.logStreamName,
                },
                encryptionConfiguration: {
                    noEncryptionConfig: 'NoEncryption',
                },
                errorOutputPrefix:
                    'raw_firehose_errors/!{firehose:error-output-type}/!{timestamp:yyyy-MM-dd}/',
                prefix: 'kinesis_delivered/IngestedDate=!{timestamp:yyyy-MM-dd}/!{firehose:random-string}/',
                dataFormatConversionConfiguration: {
                    enabled: true,
                    inputFormatConfiguration: {
                        deserializer: {
                            openXJsonSerDe: {
                                caseInsensitive: true,
                            },
                        },
                    },
                    outputFormatConfiguration: {
                        serializer: {
                            parquetSerDe: {
                                compression: 'SNAPPY',
                                writerVersion: 'V1',
                            },
                        },
                    },
                    schemaConfiguration: {
                        catalogId: cdk.Aws.ACCOUNT_ID,
                        databaseName: glueDb.databaseName,
                        tableName: tableName,
                        versionId: 'LATEST',
                        roleArn: deliveryRole.roleArn,
                    },
                },
            },
        });
        delivery.node.addDependency(deliveryRole);

        new cdk.CfnOutput(this, 'KinesisTestingBucket', { value: bucket.bucketName });
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new cdk.App();

new KinesisTestStack(app, 'kinesis-test-stack-dev', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();