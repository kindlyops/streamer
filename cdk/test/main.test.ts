import '@aws-cdk/assert/jest';
import { App } from '@aws-cdk/core';
import { KinesisTestStack } from '../src/main';

test('Snapshot', () => {
  const app = new App();
  const stack = new KinesisTestStack(app, 'test');

  expect(stack).toHaveResource('AWS::S3::Bucket');
  expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
});