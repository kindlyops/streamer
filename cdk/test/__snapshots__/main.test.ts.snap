// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Outputs": Object {
    "KinesisTestingBucket": Object {
      "Value": Object {
        "Ref": "FirehoseDeliveryBucketEBA32FCD",
      },
    },
  },
  "Parameters": Object {
    "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392ArtifactHashE56CD69A": Object {
      "Description": "Artifact hash for asset \\"4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392\\"",
      "Type": "String",
    },
    "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3BucketBF7A7F3F": Object {
      "Description": "S3 bucket for asset \\"4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392\\"",
      "Type": "String",
    },
    "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3VersionKeyFAF93626": Object {
      "Description": "S3 key for asset version \\"4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "ApplogsGlueTable5EEC7CE7": Object {
      "Properties": Object {
        "CatalogId": Object {
          "Ref": "AWS::AccountId",
        },
        "DatabaseName": Object {
          "Ref": "KinesisTestGlueDatabase7A9EF05C",
        },
        "TableInput": Object {
          "Description": "test_glue_table generated by CDK",
          "Name": "test_glue_table",
          "Parameters": Object {
            "classification": "parquet",
            "has_encrypted_data": false,
          },
          "PartitionKeys": Array [
            Object {
              "Name": "ingesteddate",
              "Type": "string",
            },
          ],
          "StorageDescriptor": Object {
            "Columns": Array [
              Object {
                "Name": "foo",
                "Type": "string",
              },
              Object {
                "Name": "n",
                "Type": "int",
              },
              Object {
                "Name": "baz",
                "Type": "int",
              },
            ],
            "Compressed": false,
            "InputFormat": "org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat",
            "Location": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "s3://",
                  Object {
                    "Ref": "FirehoseDeliveryBucketEBA32FCD",
                  },
                  "/kinesis_delivered/",
                ],
              ],
            },
            "OutputFormat": "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat",
            "SerdeInfo": Object {
              "SerializationLibrary": "org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe",
            },
            "StoredAsSubDirectories": false,
          },
          "TableType": "EXTERNAL_TABLE",
        },
      },
      "Type": "AWS::Glue::Table",
    },
    "BackupsLogStream441EBD89": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "FirehoseLogGroup1B45149B",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": Object {
      "DependsOn": Array [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3BucketBF7A7F3F",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3VersionKeyFAF93626",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3VersionKeyFAF93626",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Lambda function for auto-deleting objects in ",
              Object {
                "Ref": "FirehoseDeliveryBucketEBA32FCD",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DeliveryRole2A8B8F29": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Condition": Object {
                "StringEquals": Object {
                  "sts:ExternalId": Object {
                    "Ref": "AWS::AccountId",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "DeliveryRoleDefaultPolicyEB7AF3C1": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kinesis:DescribeStreamSummary",
                "kinesis:GetRecords",
                "kinesis:GetShardIterator",
                "kinesis:ListShards",
                "kinesis:SubscribeToShard",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "IngestStreamB5EAAE8B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "kinesis:DescribeStream",
                "kinesis:GetShardIterator",
                "kinesis:GetRecords",
                "kinesis:ListShards",
                "firehose:DeleteDeliveryStream",
                "firehose:PutRecord",
                "firehose:PutRecordBatch",
                "firehose:UpdateDestination",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "IngestStreamB5EAAE8B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "glue:GetTable",
                "glue:GetTableVersion",
                "glue:GetTableVersions",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "s3:AbortMultipartUpload",
                "s3:GetBucketLocation",
                "s3:GetObject",
                "s3:ListBucket",
                "s3:ListBucketMultipartUploads",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "FirehoseDeliveryBucketEBA32FCD",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
                Object {
                  "Fn::GetAtt": Array [
                    "FirehoseDeliveryBucketEBA32FCD",
                    "Arn",
                  ],
                },
              ],
            },
            Object {
              "Action": "logs:PutLogEvents",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:/aws/kinesisfirehose/kdf-test_glue_table:log-stream:*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeliveryRoleDefaultPolicyEB7AF3C1",
        "Roles": Array [
          Object {
            "Ref": "DeliveryRole2A8B8F29",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "FirehoseDelivery": Object {
      "DependsOn": Array [
        "DeliveryRoleDefaultPolicyEB7AF3C1",
        "DeliveryRole2A8B8F29",
      ],
      "Properties": Object {
        "DeliveryStreamType": "KinesisStreamAsSource",
        "ExtendedS3DestinationConfiguration": Object {
          "BucketARN": Object {
            "Fn::GetAtt": Array [
              "FirehoseDeliveryBucketEBA32FCD",
              "Arn",
            ],
          },
          "BufferingHints": Object {
            "IntervalInSeconds": 60,
          },
          "CloudWatchLoggingOptions": Object {
            "Enabled": true,
            "LogGroupName": Object {
              "Ref": "FirehoseLogGroup1B45149B",
            },
            "LogStreamName": Object {
              "Ref": "FirehoseLogStream2472961B",
            },
          },
          "DataFormatConversionConfiguration": Object {
            "Enabled": true,
            "InputFormatConfiguration": Object {
              "Deserializer": Object {
                "OpenXJsonSerDe": Object {
                  "CaseInsensitive": true,
                },
              },
            },
            "OutputFormatConfiguration": Object {
              "Serializer": Object {
                "ParquetSerDe": Object {
                  "Compression": "SNAPPY",
                  "WriterVersion": "V1",
                },
              },
            },
            "SchemaConfiguration": Object {
              "CatalogId": Object {
                "Ref": "AWS::AccountId",
              },
              "DatabaseName": Object {
                "Ref": "KinesisTestGlueDatabase7A9EF05C",
              },
              "RoleARN": Object {
                "Fn::GetAtt": Array [
                  "DeliveryRole2A8B8F29",
                  "Arn",
                ],
              },
              "TableName": "test_glue_table",
              "VersionId": "LATEST",
            },
          },
          "EncryptionConfiguration": Object {
            "NoEncryptionConfig": "NoEncryption",
          },
          "ErrorOutputPrefix": "raw_firehose_errors/!{firehose:error-output-type}/!{timestamp:yyyy-MM-dd}/",
          "Prefix": "kinesis_delivered/IngestedDate=!{timestamp:yyyy-MM-dd}/!{firehose:random-string}/",
          "RoleARN": Object {
            "Fn::GetAtt": Array [
              "DeliveryRole2A8B8F29",
              "Arn",
            ],
          },
          "S3BackupConfiguration": Object {
            "BucketARN": Object {
              "Fn::GetAtt": Array [
                "FirehoseDeliveryBucketEBA32FCD",
                "Arn",
              ],
            },
            "CloudWatchLoggingOptions": Object {
              "Enabled": true,
              "LogGroupName": Object {
                "Ref": "FirehoseLogGroup1B45149B",
              },
              "LogStreamName": Object {
                "Ref": "BackupsLogStream441EBD89",
              },
            },
            "ErrorOutputPrefix": "raw_backups_errors/!{firehose:error-output-type}/!{timestamp:yyyy-MM-dd}/",
            "Prefix": "raw_original_records/IngestedDate=!{timestamp:yyyy-MM-dd}/",
            "RoleARN": Object {
              "Fn::GetAtt": Array [
                "DeliveryRole2A8B8F29",
                "Arn",
              ],
            },
          },
          "S3BackupMode": "Enabled",
        },
        "KinesisStreamSourceConfiguration": Object {
          "KinesisStreamARN": Object {
            "Fn::GetAtt": Array [
              "IngestStreamB5EAAE8B",
              "Arn",
            ],
          },
          "RoleARN": Object {
            "Fn::GetAtt": Array [
              "DeliveryRole2A8B8F29",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::KinesisFirehose::DeliveryStream",
    },
    "FirehoseDeliveryBucketAutoDeleteObjectsCustomResource8971384B": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "FirehoseDeliveryBucketPolicy4B4529AC",
      ],
      "Properties": Object {
        "BucketName": Object {
          "Ref": "FirehoseDeliveryBucketEBA32FCD",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "FirehoseDeliveryBucketEBA32FCD": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "FirehoseDeliveryBucketPolicy4B4529AC": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "FirehoseDeliveryBucketEBA32FCD",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": "*",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "FirehoseDeliveryBucketEBA32FCD",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "FirehoseDeliveryBucketEBA32FCD",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "FirehoseDeliveryBucketEBA32FCD",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "FirehoseDeliveryBucketEBA32FCD",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "FirehoseLogGroup1B45149B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": "/aws/kinesisfirehose/kdf-test_glue_table",
        "RetentionInDays": 7,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "FirehoseLogStream2472961B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "FirehoseLogGroup1B45149B",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Delete",
    },
    "IngestStreamB5EAAE8B": Object {
      "Properties": Object {
        "Name": "kinesis-test-ingest-stream",
        "RetentionPeriodHours": 24,
        "ShardCount": 1,
        "StreamEncryption": Object {
          "EncryptionType": "KMS",
          "KeyId": "alias/aws/kinesis",
        },
      },
      "Type": "AWS::Kinesis::Stream",
    },
    "KinesisTestGlueDatabase7A9EF05C": Object {
      "Properties": Object {
        "CatalogId": Object {
          "Ref": "AWS::AccountId",
        },
        "DatabaseInput": Object {
          "Name": "kinesis_test",
        },
      },
      "Type": "AWS::Glue::Database",
    },
  },
}
`;
